lib/adapter.js

```diff
@@ -0,0 +1,469 @@
+var appmetrics = require('appmetrics');
+var os = require('os');
+var path = require('path');
+
+appmetrics.enable('eventloop');
+var amapi = appmetrics.monitor();
+
+// Global objects to store metric data
+var eventLoopMetrics = {
+  count : 0,
+  averageRunningTotal : 0
+};
+var cpuMetrics = {
+  system : 0,
+  user : 0,
+  count : 0
+};
+var httpMetrics = {
+  count : 0
+};
+var probeList =
+    ['leveldown', 'memcached', 'mongo', 'mysql', 'oracle', 'oracledb',
+        'postgres', 'redis', 'socketio', 'strong-oracle'];
+var probeMetrics = {};
+var heapMetrics = {
+  total : 0,
+  used : 0,
+  count : 0
+};
+var gcMetrics = {
+  used : 0,
+  count : 0
+};
+var messageMetrics = {
+  sent : 0,
+  received : 0
+};
+
+// Keep track of Cpu Profiling
+var profiling = false;
+var objectTrackingEnabled = false;
+var lastObjectHistogram = {};
+
+// Callback interval
+var callbackInterval = process.env.STRONGLOOP_BASE_INTERVAL || 60000;
+var objectTrackingCallbackInterval =
+    process.env.STRONGLOOP_OBJECT_TRACKING_INTERVAL || 15000;
+
+// Public functions to be exported
+
+// Metrics API Use function
+function use(callback) {
+  // Set up listeners for all the metrics here
+  initialise();
+
+  // The callback needs to fire every 60 seconds with data from the last 60
+  // seconds
+  setInterval(function() {
+    // Get data from last 60 seconds
+    var eventLoopData = getEventLoopData();
+    var cpuData = getCpuData();
+    var httpData = getHTTPData();
+    var heapData = getHeapData();
+    var usedHeap = getGCHeapData();
+    var messageData = getMessageData();
+
+    // Fire callback with data to match the strong-agent docs
+    callback('loop.count', eventLoopData.count);
+    callback('loop.minimum', eventLoopData.minimum);
+    callback('loop.maximum', eventLoopData.maximum);
+    callback('loop.average', eventLoopData.average);
+
+    callback('cpu.system', cpuData.system);
+    callback('cpu.total', cpuData.total);
+    callback('cpu.user', cpuData.user);
+
+    callback('http.connection.count', httpData.count);
+
+    // Callback with probe data of each probe if there have been events
+    var probeData;
+
+    for ( var i in probeList) {
+      if (probeMetrics[probeList[i]]) {
+        probeData = getProbeData(probeList[i]);
+        callback(probeList[i] + '.count', probeData.count);
+        callback(probeList[i] + '.average', probeData.average);
+        callback(probeList[i] + '.minimum', probeData.minimum);
+        callback(probeList[i] + '.maximum', probeData.maximum);
+      }
+    }
+
+    if (usedHeap > 0) {
+      callback('gc.heap.used', usedHeap);
+    }
+
+    callback('heap.total', heapData.total);
+    callback('heap.used', heapData.used);
+
+    if (messageData !== null) {
```

Comment:
`messageData` is never `null` after a recent refactor, it is always `{}` at least, perhaps with some properties, so I think the message in/out counts will be reported as `undefined` often.